CHANGES.rst

```diff
@@ -38,6 +38,9 @@ Minor changes
   which can be used to specify where to save and load from datasets.
   :pr:`432` by :user:`Lilian Boulard <LilianBoulard>`
 
+* The :class:`SuperVectorizer`'s default `OneHotEncoder` for low cardinality categorical variables now defaults to `handle_unknown="ignore"` and `drop=None` instead of `handle_unknown="error"` and `drop=if_binary`. This means
+  that categories seen only at test time will be encoded by a vector of zeroes instead of raising an error, and that no category will be dropped. :pr:`473` by :user:`Leo Grinsztajn <LeoGrin>`
```

Comment:
Why did we get rid of "drop=if_binary"? It had been added as a specific request to avoid having too big encoded matrices.

Reply:
I understand that the semantics of the two options somewhat clash (maybe something to discuss with the scikit-learn team) and bad things will happen if we have binary and unknown in the same column, but it is quite unlikely.

Reply:
Potential solutions to keep using drop="binary":
- keep handle_unknown="ignore" for non-binary columns, but use handle_unknown="error" for binary columns
- create a subclass of OneHotEncoder which maps new columns encountered during transform to "infrequent" (quite similar to `infrequent_if_exists` but without needing sklearn > 1.1).

Tagging some scikit-learn people to have other opinions: @glemaitre @Vincent-Maladiere

Reply:
I think this is related to: https://github.com/scikit-learn/scikit-learn/issues/18072 and resolved in https://github.com/scikit-learn/scikit-learn/pull/19041.

I think that the argument taken was the following: https://github.com/scikit-learn/scikit-learn/issues/18072#issuecomment-723934468

In the end, we are lenient and we are not capable of inverting properly the array.

Reply:
I think that being lenient is fine for our purposes (the case is going to be very infrequent) and thus we should just turn on both options.

Thanks @glemaitre !

Reply:
Thanks @glemaitre ! I also think that that's the best solution. The only issue is that sklearn is lenient from 0.24.2, so we get an error for earlier version (from 0.23.0). I think we can just catch it.

Reply:
What I finally chose is to just turn on both options, but to use `handle_ignore="error"` if sklearn version is < 0.24.2, while issuing a warning. I know that we excluded using `infrequent_if_exists` because changing behaviour for different  sklearn version might be confusing for users, but I figured it would be okay in this case, as it only concerns sklearn versions between 0.23 and 0.24.2.